components/ruby/lib/chef-licensing/license_key_fetcher/license_file/base.rb

```diff
@@ -0,0 +1,33 @@
+module ChefLicensing
+  module LicenseFile
+    class Base
+      def self.verify_structure(data, expected_structure)
+        return false unless data.is_a?(Hash)
+
+        expected_structure.each do |key, value|
+          return false unless data.key?(key)
+
+          if value.is_a?(Hash)
+            return false unless verify_structure(data[key], value)
+          elsif value.is_a?(Array)
+            return false unless data[key].is_a?(Array)
+
+            data[key].each do |item|
+              return false unless verify_structure(item, value[0])
+            end
+          elsif value.is_a?(Class)
+            return false unless data[key].is_a?(value)
+          else
+            return false unless data[key] == value
+          end
+        end
+
+        true
```

Comment:
I am just concerned that this will increase overhead.

Reply:
Are you suggesting optimizing the verification or remove it?

Reply:
Somehow if we can optimize to just check structure that is only the keys that might reduce the overhead. I left this comment since I felt this is something we need to keep maintaining in the future.